sdk/core/src/services/WalletStorageService.ts

```diff
@@ -293,93 +94,101 @@ export default class WalletStorageService {
     return signedCredentials as SignedCredential[]
   }
 
-  private static _getPaginationOptionsWithDefault(
-    fetchCredentialsPaginationOptions?: FetchCredentialsPaginationOptions,
-  ): PaginationOptions {
-    const { skip, limit } = fetchCredentialsPaginationOptions || {}
+  public async saveCredentials(credentials: any[], storageRegion?: string) {
+    storageRegion = storageRegion || this._storageRegion
 
-    return {
-      skip: skip || 0,
-      limit: limit || 100,
-    }
+    const responses = await this._affinidiVaultStorageService.saveCredentials(credentials, storageRegion)
+
+    return responses
   }
 
-  async encryptAndSaveCredentials(data: unknown[], storageRegion?: string) {
-    const publicKeyBuffer = this._keysService.getOwnPublicKey()
-    const encryptedCredentials = []
+  public async getAllCredentials(storageRegion?: string) {
+    storageRegion = storageRegion || this._storageRegion
 
-    for (const item of data) {
-      const cyphertext = await this._platformEncryptionTools.encryptByPublicKey(publicKeyBuffer, item)
-      encryptedCredentials.push(cyphertext)
-    }
+    const credentials = await this._affinidiVaultStorageService.searchCredentials([], storageRegion)
+
+    // should be deleted during migration to affinidi-vault Phase #2
+    const bloomCredentials = await this._bloomVaultStorageService.searchCredentials([], storageRegion)
 
-    return await this.saveCredentials(encryptedCredentials, storageRegion)
+    return [...credentials, ...bloomCredentials]
   }
 
-  async fetchAllDecryptedCredentials() {
-    const privateKeyBuffer = this._keysService.getOwnPrivateKey()
-    const allBlobs = await this.fetchAllBlobs()
-    const allCredentials = []
-    for (const blob of allBlobs) {
-      const credential = await this._platformEncryptionTools.decryptByPrivateKey(privateKeyBuffer, blob.cyphertext)
-      allCredentials.push(credential)
-    }
+  public async getCredentialsByTypes(types: string[][], storageRegion?: string) {
+    storageRegion = storageRegion || this._storageRegion
 
-    return allCredentials
-  }
+    const credentials = await this._affinidiVaultStorageService.searchCredentials(types, storageRegion)
 
-  async fetchDecryptedCredentials(fetchCredentialsPaginationOptions: FetchCredentialsPaginationOptions) {
-    const privateKeyBuffer = this._keysService.getOwnPrivateKey()
-    const blobs = await this.fetchEncryptedCredentials(fetchCredentialsPaginationOptions)
-    const credentials = []
-    for (const blob of blobs) {
-      const credential = await this._platformEncryptionTools.decryptByPrivateKey(privateKeyBuffer, blob.cyphertext)
-      credentials.push(credential)
-    }
+    // should be deleted during migration to affinidi-vault Phase #2
+    const bloomCredentials = await this._bloomVaultStorageService.searchCredentials(types, storageRegion)
 
-    return credentials
+    return [...credentials, ...bloomCredentials]
```

Comment:
ADR mentions merging the both results using a `Set<string>` instead of array concat to remove duplicate records.

Reply:
that is cannot be done with `Set<string>` here as credentials already encrypted and are parsed objects here. I don't really want make stringifying/parsing here to make them unique. Can we distinct them by `id`?

Reply:
Made this making stringify for each credential and putting it in `map` to check uniqueness.